File: test-api-pkg/ts/test-docs-overview.ts
/**
 * Mirrors website/public/sdk/overview.md against @imbrace/sdk@1.0.4 (npm)
 * — API-key auth. overview.md only has 1 testable snippet (init + getMe).
 *
 * Note: overview.md still mentions `client.activepieces` in its namespaces
 * table — SDK 1.0.4 only exposes `client.workflows`. Doc-gap flagged in
 * test-docs-workflows.ts.
 */
import "dotenv/config"
import { ImbraceClient } from "@imbrace/sdk"

const apiKey         = process.env.IMBRACE_API_KEY
const organizationId = process.env.IMBRACE_ORGANIZATION_ID
const baseUrl        = process.env.IMBRACE_GATEWAY_URL ?? "https://app-gatewayv2.imbrace.co"

if (!apiKey || !organizationId) {
  console.error("Missing IMBRACE_API_KEY or IMBRACE_ORGANIZATION_ID")
  process.exit(1)
}

const client = new ImbraceClient({ apiKey, organizationId, baseUrl, timeout: 8_000 })

let pass = 0, fail = 0
const fails: string[] = []

async function step(label: string, fn: () => Promise<unknown>) {
  process.stdout.write(`  • ${label} ... `)
  try {
    const r = await fn()
    console.log(`✓ ${JSON.stringify(r ?? {}).slice(0, 90)}`); pass++
  } catch (err: any) {
    console.log(`✗ [${err?.statusCode ?? err?.message ?? "ERR"}]`); fail++
    fails.push(`${label} → ${err?.statusCode ?? err?.message}`)
  }
}

console.log(`\n━━━ DOCS: overview.md — auth: API KEY (npm @imbrace/sdk@1.0.4) ━━━\n`)
console.log("══ §Hello world — platform.getMe ══")
await step("client.platform.getMe()", () => client.platform.getMe())

console.log(`\n━━━ Summary (overview / API key) ━━━`)
console.log(`pass=${pass}  fail=${fail}  skip=0`)
if (fails.length) { console.log("Failures:"); fails.forEach(f => console.log(`  - ${f}`)) }
process.exit(fail > 0 ? 1 : 0)

File: test-api-pkg/py/test_docs_overview.py
"""Mirrors website/public/sdk/overview.md against `imbrace==1.0.4`
(PyPI) — API-key auth. overview.md only has 1 testable Py snippet.
"""
from __future__ import annotations
import os, sys, json
from dotenv import load_dotenv
load_dotenv()
from imbrace import ImbraceClient

API_KEY = os.environ.get("IMBRACE_API_KEY"); ORG_ID = os.environ.get("IMBRACE_ORGANIZATION_ID")
GATEWAY = os.environ.get("IMBRACE_GATEWAY_URL", "https://app-gatewayv2.imbrace.co")
if not API_KEY or not ORG_ID:
    print("Missing IMBRACE_API_KEY or IMBRACE_ORGANIZATION_ID"); sys.exit(1)

passed = failed = 0; failures: list[str] = []


def step(label, fn):
    global passed, failed
    sys.stdout.write(f"  - {label} ... "); sys.stdout.flush()
    try:
        r = fn()
        print(f"OK {json.dumps(r, default=str)[:90]}"); passed += 1
    except Exception as e:
        print(f"FAIL [{str(e)[:80]}]"); failed += 1
        failures.append(f"{label} -> {str(e)[:80]}")


print(f"\n=== DOCS: overview.md - auth: API KEY (PyPI imbrace==1.0.4) ===\n")

print("== §Hello world — context manager + platform.get_me ==")


def _hello():
    with ImbraceClient(api_key=API_KEY, organization_id=ORG_ID, gateway=GATEWAY, timeout=8) as c:
        return c.platform.get_me()


step("with ImbraceClient(api_key=...) as c: c.platform.get_me()", _hello)

print(f"\n=== Summary (overview / API key) ===")
print(f"pass={passed}  fail={failed}  skip=0")
if failures:
    print("Failures:")
    for f in failures: print(f"  - {f}")
sys.exit(1 if failed > 0 else 0)
